examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py

```diff
@@ -43,98 +44,172 @@ class TaskMapping(NamedTuple):
     group: Optional[str] = None
 
 
-def assets_defs_from_airflow_instance(
-    airflow_webserver_url: str,
-    auth: Tuple[str, str],
-    instance_name: str,
-    task_maps: Sequence[TaskMapping] = [],
-) -> List[AssetsDefinition]:
-    api_url = f"{airflow_webserver_url}/api/v1"
-    dag_infos: List[DagInfo] = []
-
-    # First, we attempt to fetch all the DAGs present in the Airflow instance.
-    response = requests.get(f"{api_url}/dags", auth=auth)
-    if response.status_code == 200:
-        dags = response.json()
-        for dag in dags["dags"]:
-            dag_infos.append(
+class AirflowAuthBackend(ABC):
+    def get_session(self) -> requests.Session:
+        raise NotImplementedError("This method must be implemented by subclasses.")
+
+
+class BasicAuthBackend(AirflowAuthBackend):
+    def __init__(self, username: str, password: str):
+        self.username = username
+        self.password = password
+
+    def get_session(self) -> requests.Session:
+        session = requests.Session()
+        session.auth = (self.username, self.password)
+        return session
+
+
+class AirflowInstance(NamedTuple):
```

Comment:
seems a little goofy to have this be a NamedTuple but nbd

Reply:
what's goofy about it being a namedtuple? I guess just vs a regular class you mean?

Honestly it's just learned behavior from the codebase at this point

Reply:
Generally NamedTuple are "dumb" value objects that represent a value/record at a point in time. This is a more complex object that will likely accumulate state, require caching, that sort of thing.